Dockerfile

```diff
@@ -1,5 +1,5 @@
 # Builder image
-FROM golang:1.10.1-alpine as builder
+FROM golang:1.11-alpine as builder
```

Comment:
More specific is more reproducile. `1.11.2` is better here.

Reply:
`1.11` always updates to latest `1.11.x` version

Reply:
If I build this Dockefile today, and tomorrow 1.11.3 comes out, I will continue using 1.11.2. Someone else that starts building tomorrow will be on 1.11.3. So it has some convenience but it is less reproducible.

Reply:
Sorry, if that isn't clear, Docker does not auto-update tags once an image has been downloaded.

Reply:
I understand the go build version is determined when building image. I think we should make clear where to use this `Dockerfile`, is it used for developer quickly build a TiDB image or used in `Docker hub` or somewhere else.
If it is used for developer building, use the latest go version may be better.

Reply:
I agree that it can be good to get some developers testing out compatibility of newer version of dependencies. But I think this is already happening a lot from those compiling with their own go compiler. I use the Dockerfile to quickly build an image for testing and I want it to be as reproducible as possible.
I also agree its not that a big of a deal since it is clear from this breakage that this file is not used much. And the workflow is still broken because it download the vendor modules every time.